bot/src/keepers/debt.ts

```diff
@@ -0,0 +1,153 @@
+            console.info(
+                `debt keeper: wallet "${walletAddress}" won auction "${auction.id}", moving on to collection`
+            );
+            await collectDebtAuction(network, auction.id);
+            console.info(`debt keeper: auction "${auction.id}" was successfully collected`);
+            return;
+        } else {
+            console.info(
+                `debt keeper: auction "${auction.id}" has already finished and the winner is not "${walletAddress}"`
+            );
+            return;
+        }
+    }
+    if (auctionTransaction.state === 'requires-restart') {
+        console.info(`debt keeper: auction "${auction.id}" is inactive`);
+        return;
+    }
+
+    // check auction's net profit – profit without transaction fees
+    const netProfit = auctionTransaction.marketUnitPrice
+        .minus(auctionTransaction.unitPrice)
+        .times(auctionTransaction.nextMaximumLotReceived)
+        .minus(auctionTransaction.combinedBidFeesDai);
+    if (netProfit.isLessThan(new BigNumber(KEEPER_DEBT_MINIMUM_NET_PROFIT_DAI))) {
+        console.info(
+            `debt keeper: auction "${
+                auction.id
+            }" net profit is smaller than min profit (${formatToAutomaticDecimalPointsString(
+                netProfit
+            )} < ${KEEPER_DEBT_MINIMUM_NET_PROFIT_DAI})`
+        );
+        return;
+    } else {
+        console.info(
+            `debt keeper: auction "${auction.id}" net profit is ${formatToAutomaticDecimalPointsString(
+                netProfit
+            )} DAI after transaction fees, checking wallet DAI balance`
+        );
+    }
+
+    // check the VAT DAI balance
+    const vatBalanceDai = await fetchVATbalanceDAI(network, walletAddress);
+    if (vatBalanceDai.isLessThan(auctionTransaction.bidAmountDai)) {
+        console.info(
+            `debt keeper: VAT DAI balance (${formatToAutomaticDecimalPointsString(
+                vatBalanceDai
+            )}) is less than fixed bid amount (${formatToAutomaticDecimalPointsString(
+                auctionTransaction.bidAmountDai
+            )}) for auction "${auction.id}", checking walllet DAI balance to deposit to VAT`
+        );
+        // check the wallet's DAI balance
+        const balanceDai = await fetchBalanceDAI(network, walletAddress);
+        if (balanceDai.isLessThan(auctionTransaction.bidAmountDai)) {
+            console.info(
+                `debt keeper: wallet DAI balance (${formatToAutomaticDecimalPointsString(
+                    balanceDai
+                )}) is less than fixed bid amount (${formatToAutomaticDecimalPointsString(
+                    auctionTransaction.bidAmountDai
+                )}) for auction "${auction.id}"`
+            );
+        } else {
+            console.info('debt keeper: wallet DAI balance is within limits, depositing to VAT');
+            await depositToVAT(network, walletAddress, auctionTransaction.bidAmountDai);
```

Comment:
Don't you need to check allowance before depositing to VAT otherwise it will fail?